src/components/managers/managers.js

```diff
@@ -0,0 +1,64 @@
+//@flow
+import {TimelineManager} from './timeline-manager';
+import {UIManager} from '../../index';
+import getLogger from '../../utils/logger';
+
+/**
+ * @class CuePoint
+ */
+class Managers {
+  static _logger: any;
+  _managerRegistry: Map<(name: string) => Object | null> = new Map();
+
+  /**
+   * @constructor
+   * @param {UIManager} uiManager - The UI manager.
+   * @param {any} store - The store.
+   */
+  constructor(uiManager: UIManager, store: any) {
+    Managers._logger = getLogger('Managers');
+    this.registerManager('timeline', new TimelineManager(uiManager, store));
+  }
+
+  /**
+   * @param {string} name - the manager name
+   * @param {Object} manager - the manager object
+   * @returns {void}
+   */
+  registerManager(name: string, manager: Object): void {
+    if (this._managerRegistry.has(name)) {
+      Managers._logger.debug(`${name} manager already exists`);
+    } else {
+      this._managerRegistry.set(name, manager);
+      Managers._logger.debug(`${name} manager registered`);
+    }
+  }
+
+  /**
+   *
+   * @param {string} name - the manager name
+   * @returns {Object} - the manager object
+   */
+  getManager(name: string): Object | void {
```

Comment:
Create enums for the names and type it.

Reply:
the `managers` is open to expanding by register manager from outside. this couldn't be in enum.